linera-core/src/client/mod.rs

```diff
@@ -2480,35 +2484,22 @@ where
             .already_handled_proposal(round, &executed_block.block);
         let key_pair = self.key_pair().await?;
         // Create the final block proposal.
-        let (proposal, is_locking) = if let Some(locking) = info.manager.requested_locking {
-            let proposal = Box::new(match *locking {
+        let proposal = if let Some(locking) = info.manager.requested_locking {
+            Box::new(match *locking {
                 LockingBlock::Regular(cert) => BlockProposal::new_retry(round, cert, &key_pair),
                 LockingBlock::Fast(proposal) => {
                     BlockProposal::new_initial(round, proposal.content.block, &key_pair)
                 }
-            });
-            (proposal, true)
+            })
         } else {
             let block = executed_block.block.clone();
-            let proposal = Box::new(BlockProposal::new_initial(round, block, &key_pair));
-            (proposal, false)
+            Box::new(BlockProposal::new_initial(round, block, &key_pair))
         };
         if !already_handled_locally {
-            let local_node = &self.client.local_node;
             // Check the final block proposal. This will be cheaper after #1401.
             if let Err(err) = local_node.handle_block_proposal(*proposal.clone()).await {
                 match err {
-                    LocalNodeError::BlobsNotFound(blob_ids) => {
-                        let blobs = if is_locking {
-                            local_node
-                                .get_locking_blobs(&blob_ids, self.chain_id)
-                                .await?
-                                .ok_or_else(|| {
-                                    ChainClientError::InternalError("Missing local locking blobs")
-                                })?
-                        } else {
-                            self.pending_blobs().iter().cloned().collect()
-                        };
+                    LocalNodeError::BlobsNotFound(_) => {
```

Comment:
How is it that we no longer need the code from the left?

Reply:
We know which blobs we need based on whether we are re-proposing a locking block or making a new proposal. So we fetch those in line 2453. (These are probably exactly the blobs in the chain manager, so there's room for more simplification, but this PR is already getting a bit large.)

Reply:
Thanks.